Drv/TcpClient/test/ut/TcpClientTester.cpp

```diff
@@ -84,12 +84,21 @@ void TcpClientTester ::test_with_loop(U32 iterations, bool recv_thread) {
             }
         }
         // Properly stop the client on the last iteration
-        if ((1 + i) == iterations && recv_thread) {
-            this->component.stop();
-            this->component.join();
+        if (recv_thread) {
+            //this->component.close();
+            if ((1 + i) == iterations) {
+                this->component.m_task_lock.lock();
```

Comment:
These locks should be done in the shutdown and close methods, not just the UT.

Reply:
I moved the mutex in the close/shutdown/stop functions. I had to update the `TcpServerTester.cpp` a bit for it to work: the client needs to be closed after the server is closed. I do not have the explanation for this behavior, but otherwise I was getting the following warning:

```
Connected to 127.0.0.1:57755 as a tcp client
Accepted client at 127.0.0.1:57755
[WARNING] Failed to recv from port with status -8 and errno 0
```

There was a comment stating: ` // Client must be closed first or the server risks binding to an existing address`. Let me know if it is a problem.

Reply:
I also added the muted on the stop function. Can you confirm that it is something that is expected?